workers/unity/Packages/com.improbable.gdk.core/Collections/ReceivedMessageListExtensions.cs

```diff
@@ -1,13 +1,48 @@
+using Improbable.Gdk.Core.Commands;
+
 namespace Improbable.Gdk.Core
 {
     // todo consider exposing the private methods - the upper limit one can be used to find the last event/auth change
     internal static class ReceivedMessageListExtensions
     {
+        // binary search for the command for given request ID
+        // invariant: lower <= target <= upper
+        public static int GetResponse<T>(this ReceivedMessageList<T> list, long requestId)
```

Comment:
Would this be better as an `internal` function?

Reply:
I believe the class scope makes this internal. But it's also public function on an internal collection.